lib/bundler/settings.rb

```diff
@@ -26,13 +27,31 @@ def [](name)
       end
     end
 
+    # In Bundler pre 2.0 any command ever set was stored in the local config
+    # file. In Bundler 2.0 they are only saved for that current command and
+    # `bundle config <setting name> <setting value>` is used for setting
+    # remembered arguments.
+    # See https://trello.com/c/yGsPNDpg
     def []=(key, value)
+      @use_current ||= (Bundler::VERSION.split(".")[0].to_i >= 2)
```

Comment:
Because `Bundler::VERSION` is less than 2 (https://github.com/pducks32/bundler/blob/8a8b1513e4c1651f39c5361e1d5d93a41411a202/lib/bundler/version.rb#L5), the behavior of bundler isn't changed by this PR.